core/funcs.go

```diff
@@ -852,6 +896,150 @@ func init() {
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslFloatT, RslIntT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				arg := f.args[0]
+				val := arg.value.RequireFloatAllowingInt(f.i, arg.node)
+				return newRslValues(f.i, f.callNode, math.Ceil(val))
+			},
+		},
+		{
+			Name:             FUNC_FLOOR,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslFloatT, RslIntT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				arg := f.args[0]
+				val := arg.value.RequireFloatAllowingInt(f.i, arg.node)
+				return newRslValues(f.i, f.callNode, math.Floor(val))
+			},
+		},
+		{
+			Name:             FUNC_MIN,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslListT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				// input is a list of numbers
+				list := f.args[0].value.RequireList(f.i, f.args[0].node)
+				if len(list.Values) == 0 {
+					f.i.errorf(f.callNode, "Cannot find minimum of empty list")
+				}
+
+				minVal := math.MaxFloat64
+				for idx, item := range list.Values {
+					val, ok := item.TryGetFloatAllowingInt()
+					if !ok {
+						f.i.errorf(f.args[0].node, "%s() requires a list of numbers, got %q at index %d", FUNC_MIN, TypeAsString(item), idx)
+					}
+					minVal = math.Min(minVal, val)
+				}
+				return newRslValues(f.i, f.callNode, minVal)
+			},
+		},
+		{
+			Name:             FUNC_MAX,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslListT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				// input is a list of numbers
+				list := f.args[0].value.RequireList(f.i, f.args[0].node)
+				if len(list.Values) == 0 {
+					f.i.errorf(f.callNode, "Cannot find maximum of empty list")
+				}
+
+				maxVal := -math.MaxFloat64
+				for idx, item := range list.Values {
+					val, ok := item.TryGetFloatAllowingInt()
+					if !ok {
+						f.i.errorf(f.args[0].node, "%s() requires a list of numbers, got %q at index %d", FUNC_MAX, TypeAsString(item), idx)
+					}
+					maxVal = math.Max(maxVal, val)
+				}
+				return newRslValues(f.i, f.callNode, maxVal)
+			},
+		},
+		{
+			Name:             FUNC_CLAMP,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslFloatT, RslIntT}, {RslFloatT, RslIntT}, {RslFloatT, RslIntT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				// input is a number and a min and max
+				val, ok := f.args[0].value.TryGetFloatAllowingInt()
+				if !ok {
+					f.i.errorf(f.args[0].node, "%s() requires a number, got %q at index %d", FUNC_CLAMP, TypeAsString(f.args[0].value), 0)
+				}
+				min := -math.MaxFloat64
```

Comment:
nitpick: I prefer to avoid shadowing other variables, so would name this something like `minVal` (and `maxVal`) instead.